src/Channels/Networking/TransportProvider.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Globalization;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Channels.Networking
+{
+    /// <summary>
+    /// Represents a transport that is able to establish connections from client to server,
+    /// or to start a listener to operate as a server
+    /// </summary>
+    public abstract class TransportProvider : IDisposable
+    {
+        private ChannelFactory channelFactory;
+
+        /// <summary>
+        /// Provides access to the channel factory for this provider
+        /// </summary>
+        /// <remarks>This factory is lazily instantiated</remarks>
+        protected ChannelFactory ChannelFactory => channelFactory ?? (channelFactory = new ChannelFactory());
+
+        /// <summary>
+        /// Open a client connection to the designated resource
+        /// </summary>
+        public abstract Task<IChannel> ConnectAsync(string configuration);
+        /// <summary>
+        /// Create a server instance listening to the designated resource
+        /// </summary>
+        public abstract Task<IDisposable> StartServerAsync(string configuration, Action<IChannel> callback);
+
+        /// <summary>
+        /// Releases all resources owned by this object
+        /// </summary>
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
+            Dispose(true);
+        }
+
+        /// <summary>
+        /// Releases all resources owned by this object
+        /// </summary>
+        protected virtual void Dispose(bool disposing)
+        {
+            if(disposing)
+            {
+                channelFactory?.Dispose();
+                channelFactory = null;
+            }
+        }
+
+        /// <summary>
+        /// Parse the configuration as an IPEndPoint
+        /// </summary>
+        public static ValueTask<IPEndPoint> ParseIPEndPoint(string configuration, int defaultPort = -1)
```

Comment:
Lets not do  parsing at this level. We take objects directly. Let another helper API parse whatever format.